src/test/rgw/bucket_notification/test_bn.py

```diff
@@ -2482,6 +2551,271 @@ def test_metadata_filter_ampq():
+    result, status = s3_notification_conf.set_config()
+    assert_equal(status/100, 2)
+
+    # get all notifications
+    result, status = s3_notification_conf.get_config()
+    assert_equal(status/100, 2)
+    for conf in result['TopicConfigurations']:
+        filter_name = conf['Filter']['Key']['FilterRules'][0]['Name']
+        assert filter_name == 'prefix' or filter_name == 'suffix' or filter_name == 'regex', filter_name
+
+    sample_space = ['hello.kaboom', 'hello.txt', 'hello123.txt', 'hello', 'world1.log', 'world2log', 'world3.log', 'hello.txt', 'hell.txt', 'worldlog.txt', 'kaboom.log', 'foobar.log', 'log', 'he123ll.txt', 'wo', 'h', 'txt', 'world.log.txt']
+    expected_in1 = [key for key in sample_space if key.startswith('hello') is False]
+    expected_in2 = [key for key in sample_space if key.endswith('log') is False]
+    expected_in3 = [key for key in sample_space if re.match(r'([a-z]+)\.txt', key) is None]
+
+    # create objects in bucket
+    for key_name in sample_space:
+        key = bucket.new_key(key_name)
+        key.set_contents_from_string('bar')
+
+    print('wait for 5sec for the messages...')
+    time.sleep(5)
+
+    found_in1 = []
+    found_in2 = []
+    found_in3 = []
+
+    print('wait for the messages...')
+    wait_for_queue_to_drain(topic_name, http_port=port)
+    # check amqp receiver
+    events = receiver.get_and_reset_events()
+
+    for event in events:
+        notif_id = event['Records'][0]['s3']['configurationId']
+        key_name = event['Records'][0]['s3']['object']['key']
+        awsRegion = event['Records'][0]['awsRegion']
+        assert_equal(awsRegion, zonegroup)
+        bucket_arn = event['Records'][0]['s3']['bucket']['arn']
+        assert_equal(bucket_arn, "arn:aws:s3:"+awsRegion+"::"+bucket_name)
+        if notif_id == notification_name+'_1':
+            found_in1.append(key_name)
+        elif notif_id == notification_name+'_2':
+            found_in2.append(key_name)
+        elif notif_id == notification_name+'_3':
+            found_in3.append(key_name)
+        else:
+            assert False, 'invalid notification: ' + notif_id
+
+    assert_equal(set(found_in1), set(expected_in1))
+    assert_equal(set(found_in2), set(expected_in2))
+    assert_equal(set(found_in3), set(expected_in3))
+
+     # delete objects
+    for key in bucket.list():
+        key.delete()
+    
+    # cleanup
+    receiver.close(task)
+    s3_notification_conf.del_config()
+    topic_conf.del_config()
+    # delete the bucket
+    conn.delete_bucket(bucket_name)
+
+# @attr('kafka_test')
+# def test_negative_filter_kafka():
+#     """ test notification of filtering metadata, kafka endpoint """
+#     conn = connection()
+#     metadata_negative_filter('kafka', conn)
+#     key_negative_filter('http', conn)
+
+@attr('http_test')
+def test_negative_filter_http():
+    """ test notification of filtering metadata, http endpoint """
+    conn = connection()
+    metadata_negative_filter('http', conn)
```

Comment:
you can split these into 2 tests:
`test_negative_metadata_filter_http()`
`test_negative_key_filter_http()`